### ISSUE TITLE
Reopening #67 Issue

### ISSUE THREAD

**Issue by sdow1ll** [evolutionaryscale/esm#75]
Hi,

There is a bit more to the discussion on #67 and using `_forward` didn't really work for me at first. Could you please review the newest comments on #67 and provide some feedback? If there is another way to get logits more nicely, I would appreciate your guidance on this matter.

Thanks!

**Comment by ebetica** [comment]
In 3.0.2, we changed the `_forward` to `logits`. https://github.com/evolutionaryscale/esm/blob/2066a2a2c6f46c5ed8934b59037891701a9dfa2f/examples/local_generate.py#L92-L104

Does this fix your issue? Currently it's only on main, but we'll push to pypi soon.

**Comment by sdow1ll** [comment]
Just updated your repo, tried it on my own based on the above:

`from esm.models.esm3 import ESM3
from esm.sdk.api import ESM3InferenceClient, ESMProtein, ESMProteinTensor, LogitsConfig, LogitsOutput
from esm.utils.structure.protein_chain import ProteinChain
from esm.utils.types import FunctionAnnotation

# Initialize the client
client = ESM3.from_pretrained("esm3_sm_open_v1")

# Get sample protein
protein = ESMProtein(
    sequence=("QVQLQQSGPGLVKPSQTLSLTCAISGDSVSSYNAVWNWIRQSPSRGLEWLGRTYYRSGWYNDYAESVKSRITINPDTSKNQFSLQLNSVTPEDTAVYYCARSGHITVFGVNVDAFDMWGQGTMVTVSS")
)
print(protein)
protein.coordinates = None
protein.function_annotations = None
protein.sasa = None

# Encode the protein
protein_tensor = client.encode(protein)

# Get the logits
logits_output = client.logits(protein_tensor, LogitsConfig(sequence=True))

# Check and print the logits
if isinstance(logits_output, LogitsOutput):
    if logits_output.logits is not None and logits_output.logits.sequence is not None:
        sequence_logits = logits_output.logits.sequence
        print("\nLogits of Sequence: \n", sequence_logits)
    else:
        print("Logits are not available.")
else:
    print(f"LogitsOutput was expected but got {logits_output}")
`

Output:

`ESMProtein(sequence='QVQLQQSGPGLVKPSQTLSLTCAISGDSVSSYNAVWNWIRQSPSRGLEWLGRTYYRSGWYNDYAESVKSRITINPDTSKNQFSLQLNSVTPEDTAVYYCARSGHITVFGVNVDAFDMWGQGTMVTVSS', secondary_structure=None, sasa=None, function_annotations=None, coordinates=None, plddt=None, ptm=None, potential_sequence_of_concern=False)

Logits of Sequence: 
 tensor([[[-19.2500, -19.2500, -19.2500,  ..., -19.2500, -19.2500, -19.2500],
         [-19.1250, -19.1250, -19.0000,  ..., -19.0000, -19.1250, -19.1250],
         [-19.5000, -19.5000, -19.3750,  ..., -19.3750, -19.5000, -19.5000],
         ...,
         [-21.6250, -21.5000, -21.6250,  ..., -21.6250, -21.6250, -21.6250],
         [-20.6250, -20.6250, -20.7500,  ..., -20.6250, -20.7500, -20.7500],
         [-15.8125, -15.8125, -15.8750,  ..., -15.8750, -15.8750, -15.8750]]],
       device='cuda:0')`
       
       This works, thank you very much for your help!

**Comment by ebetica** [comment]
Please feel free to reopen if you have additional issues!
